server/storage/schema/schema.go

```diff
@@ -64,3 +87,93 @@ func detectStorageVersion(lg *zap.Logger, tx backend.ReadTx) (*semver.Version, e
 	copied := V3_5
 	return &copied, nil
 }
+
+func migrateByOneVersion(lg *zap.Logger, tx backend.BatchTx, current semver.Version, upgrade bool) (*semver.Version, error) {
+	var target semver.Version
+	if upgrade {
+		target = semver.Version{Major: current.Major, Minor: current.Minor + 1}
+	} else {
+		target = semver.Version{Major: current.Major, Minor: current.Minor - 1}
+	}
+	migration := newMigration(lg, current, target)
+	err := migration.run(tx)
+	if err != nil {
+		return nil, err
+	}
+	return &target, nil
+}
+
+type migration struct {
+	current, target semver.Version
+	actions         []migrateAction
+}
+
+func newMigration(lg *zap.Logger, current semver.Version, target semver.Version) *migration {
+	var key semver.Version
+	// actions should be taken current higher version
+	if current.LessThan(target) {
+		key = target
+	} else {
+		key = current
+	}
+	actions, found := versionMigateActions[key]
+	if !found {
+		lg.Panic("version is not supported", zap.String("storage-version", target.String()))
+	}
+	return &migration{
+		current: current,
+		target:  target,
+		actions: actions,
+	}
+}
+
+func (m *migration) run(tx backend.BatchTx) error {
+	var err error
+	upgrade := m.current.LessThan(m.target)
+	tx.Lock()
+	defer tx.Unlock()
+	for i := 0; i < len(m.actions); i++ {
+		if upgrade {
+			err = m.actions[i].UnsafeUpgrade(tx)
+		} else {
+			// Execute downgrades in revered order
+			err = m.actions[len(m.actions)-1-i].UnsafeDowngrade(tx)
+		}
+		if err != nil {
+			return err
+		}
+	}
+	// Storage version is available since v3.6, downgrading target v3.5 should clean this field.
+	if !m.target.LessThan(V3_6) {
+		UnsafeSetStorageVersion(tx, &m.target)
+	}
+	return nil
+}
+
+var versionMigateActions = map[semver.Version][]migrateAction{
+	V3_6: {
+		&addNewFieldAction{bucket: Meta, fieldName: MetaStorageVersionName, fieldValue: []byte("") /* leaving empty as StorageVersion should be set by migrator */},
```

Comment:
How about making it a dedicated enum, such it's easy to find all places in the code that take care of providing the actual value instead of ""

Reply:
Makes sense, added a variable.